src/main/java/org/opentripplanner/updater/vehicle_parking/VehicleParkingUpdater.java

```diff
@@ -0,0 +1,189 @@
+package org.opentripplanner.updater.vehicle_parking;
+
+import org.opentripplanner.graph_builder.linking.LinkingDirection;
+import org.opentripplanner.graph_builder.linking.VertexLinker;
+import org.opentripplanner.model.FeedScopedId;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.edgetype.StreetVehicleParkingLink;
+import org.opentripplanner.graph_builder.linking.DisposableEdgeCollection;
+import org.opentripplanner.routing.edgetype.VehicleParkingEdge;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.vehicle_parking.VehicleParking;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingHelper;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingService;
+import org.opentripplanner.routing.vertextype.VehicleParkingEntranceVertex;
+import org.opentripplanner.updater.GraphUpdaterManager;
+import org.opentripplanner.updater.GraphWriterRunnable;
+import org.opentripplanner.updater.PollingGraphUpdater;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Graph updater that dynamically sets availability information on vehicle parking lots.
+ * This updater fetches data from a single {@link VehicleParkingDataSource}.
+ */
+public class VehicleParkingUpdater extends PollingGraphUpdater {
+
+    private static final Logger LOG = LoggerFactory.getLogger(VehicleParkingUpdater.class);
+
+    private GraphUpdaterManager updaterManager;
+
+    private final Map<VehicleParking, List<VehicleParkingEntranceVertex>> verticesByPark = new HashMap<>();
+
+    private final Map<VehicleParking, List<DisposableEdgeCollection>> tempEdgesByPark = new HashMap<>();
+
+    private final VehicleParkingDataSource source;
+
+    private VertexLinker linker;
+
+    private VehicleParkingService vehicleParkingService;
+
+    public VehicleParkingUpdater(VehicleParkingUpdaterParameters parameters) {
+        super(parameters);
+        // Set source from preferences
+        source = new KmlBikeParkDataSource(parameters.sourceParameters());
```

Comment:
Should this be abstracted to a SourceFactory similarly as in BikeRentalUpdater?

Reply:
There is another WIP branch to add updaters for OpenBikeBox, ParkApi which contains some refactoring of the logic:
https://github.com/realCity/OpenTripPlanner/commits/feature/vehicle-parking-updaters